canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js

```diff
@@ -4426,14 +4442,98 @@ export default class SVGCanvasRenderer {
 		this.canvasSVG.selectAll(".d3-data-link-selection-area").classed("d3-extra-width", state);
 	}
 
-	// Returns the node that is at the current mouse position. If nodeProximity
+	// Returns a node, if one can be found, at the position indicated by
+	// the clientX and clientY coordinates in the d3Event.
+	getNodeAtMousePos(d3Event) {
+		const nodeGrpElement = this.getElementWithClassAtMousePos(d3Event, "d3-node-group");
+		return this.getNodeForElement(nodeGrpElement);
+	}
+
+	// Returns a node input port ID, if one can be found, at the position
+	// indicated by the clientX and clientY coordinates in the d3Event.
+	getInputNodePortIdAtMousePos(d3Event) {
```

Comment:
Would these functions make more sense to be part of `svg-vanvas-utils-nodes/js`?

Reply:
I think I'll leave theme where they are for now. Eventually I want to refactor sv-canvas-renderer into different files.